### PR TITLE
feat: add permissions endpoint

### PR DESCRIPTION
**Pull request by e-schneid**

### CODE REVIEW

File: packages/api/src/middleware/psa.js [hunk 1]
```diff
@@ -1,67 +0,0 @@
...
-/**
- * Verify that the authenticated request is for a user who is authorized to use
- * the Pinning Service API.
- *
- * @param {Handler} handler
- * @returns {Handler}
- */
-export function withPinningAuthorized(handler) {
-  return async (event, ctx) => {
-    // TODO: we need withAuth middleware so we don't have to do this twice
```
Comment (e-schneid) [inline packages/api/src/middleware/psa.js:55]: most of the changes in this PR were for this TODO here.  I wanted to only have to fetch auth data once in a given request.

**Comment by hugomrdias** [comment]
can you check the ci error please ?

**Comment by e-schneid** [comment]
>

**Review by trigramdev9 — APPROVED** [review]

**Comment by JeffLowe** [comment]
Checked on staging to confirm full flows...
- blocked user still able to LOGIN to web application
- upon login to web UI, blocked user gets a message explaining their account was blocked
- blocked user unable to create any new api keys & see tooltip explaining
- blocked user can see list of all their API keys, User still able to DELETE api key
- blocked user unable to upload any files, user feedback in button tooltip
- blocked user still able to see their uploaded files in /files, complete copy functions, and even delete functions.
- blocked user unable to request access to the Pinning Service Allowlist, button does not show in UI

**Pull request merged**